salt/states/pkg.py

```diff
@@ -1690,45 +1690,45 @@ def installed(
             # checks reinstall targets works.
             pkg_ret = {}
 
-        if 'pkg.hold' in __salt__:
-            if 'hold' in kwargs:
-                try:
-                    if kwargs['hold']:
-                        hold_ret = __salt__['pkg.hold'](
-                            name=name, pkgs=pkgs, sources=sources
-                        )
-                    else:
-                        hold_ret = __salt__['pkg.unhold'](
-                            name=name, pkgs=pkgs, sources=sources
-                        )
-                except (CommandExecutionError, SaltInvocationError) as exc:
-                    comment.append(str(exc))
+    if 'pkg.hold' in __salt__:
+        if 'hold' in kwargs:
+            try:
+                if kwargs['hold']:
+                    hold_ret = __salt__['pkg.hold'](
+                        name=name, pkgs=desired, sources=sources
+                    )
+                else:
+                    hold_ret = __salt__['pkg.unhold'](
+                        name=name, pkgs=desired, sources=sources
+                    )
```

Comment:
Or shorter, so easier comprehend half-year later while debuggin and no same code repetitions:
```python
hold_ret = __salt__['pkg.{}hold'.format(not kwargs['hold'] and 'un' or '')](
        name=name, pkgs=desired, sources=sources
)
```

Reply:
This doesn't look that much better. It's certainly a lot less readable. It would be more readable like this:

```python
action = 'pkg.hold' if kwargs['hold'] else 'pkg.unhold'
hold_ret = __salt__[action](
    name=name, pkgs=desired, sources=sources
)
```